pkg/plugin/datasource.go

```diff
@@ -573,21 +573,24 @@ func (d *Datasource) VLAPIQuery(rw http.ResponseWriter, req *http.Request) {
 			return
 		}
 	}
+	defer resp.Body.Close()
+
 	reader := io.Reader(resp.Body)
 	if resp.Header.Get("Content-Encoding") == "gzip" {
-		reader, err = gzip.NewReader(reader)
+		gzipReader, err := gzip.NewReader(reader)
```

Comment:
You do not need to reassign reader; you can just close it on line 584
it would be one line change

Reply:
fixed it